gcc/rust/parse/rust-parse-impl.h

```diff
@@ -5784,6 +5785,7 @@ Parser<ManagedTokenSource>::parse_trait_impl_item ()
       // function or method
       return parse_trait_impl_function_or_method (visibility,
 						  std::move (outer_attrs));
+    case ASYNC:
```

Comment:
`const` is handled this way because inside a trait, you could have a const variable or a constant function:

```rust
impl Foo for u32 {
    const TRAIT_ASSOC_CST: u32 = 15;
    
    const fn constant_fn() -> u32 { 14 }
}
```

for async, we can only have `async` functions - so we don't need to bundle the cases for `CONST` and `ASYNC` together.

Instead, this is what I would like to see:

```suggestion
    case ASYNC:
        return parse_async_item (visibility, std::move(outer_attrs))
    case CONST:
         /* snip */
```
```